Cargo.toml

```diff
@@ -8,13 +8,16 @@ members = [
     "shared", 
     "agent", 
     "controller", 
+    "samples/apps/coap-device",
```

Comment:
I am not sure if this should live in akri as it is a mock device. I feel like the `coap-demo.md` is sufficient for getting people going. Or we could put it on a branch of akri, or you could host it if interested. @bfjelds for thoughts